primitives/core/src/ed25519.rs

```diff
@@ -508,15 +511,14 @@ impl TraitPair for Pair {
 
 	/// Get the public key.
 	fn public(&self) -> Public {
-		let mut r = [0u8; 32];
-		let pk = self.0.public.as_bytes();
-		r.copy_from_slice(pk);
-		Public(r)
+		// does this consume public? Is that why we used copy_from_slice?
```

Comment:
nope it won't consume public. - it can't take ownership of anything from an &self as it's read only.
(If it was a problem it wouldn't compile.)